src/test/rpc/LedgerRPC_test.cpp

```diff
@@ -494,6 +494,18 @@ class LedgerRPC_test : public beast::unit_test::suite
                 "json", "ledger", "{ \"ledger_index\" : 1000000000000000 }");
             checkErrorValue(ret, "invalidParams", "Invalid parameters.");
         }
+
+        {
+            // ask for an zero index
+            Json::Value jvParams;
+            jvParams[jss::ledger_index] = "validated";
+            jvParams[jss::index] =
+                "00000000000000000000000000000000000000000000000000000000000000"
```

Comment:
Hello, I know that this check corresponds to the case of a malformed ledger-object-ID inside the `ledger_entry` RPC call.

Are we sure that no ledger-object-ID can ever get this particular hash value? Despite using specific ledger-object prefixes, cryptography does not provide any guarantees that this particular hash is unattainable. Am I wrong?

Reply:
It doesn't provide a guarantee that 2 different objects in the same namespace will not get the same hash too.

Reply:
that is true